packages/system/src/Location/Components/Toolbar.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { FC, useEffect, useRef } from 'react';
+import {
+  useNotification,
+  DropdownMenu,
+  DropdownMenuItem,
+  useTranslation,
+  DeleteIcon,
+  useTableStore,
+  AppBarContentPortal,
+  FilterController,
+  AlertModal,
+} from '@openmsupply-client/common';
+import { useLocationDelete } from '../ListView/api';
+import { Location } from '../types';
+
+type DeleteError = {
+  locationName: string;
+  message: string;
+};
+
+export const Toolbar: FC<{
+  data: Location[];
+  filter: FilterController;
+}> = ({ data }) => {
+  const t = useTranslation('inventory');
+  const { mutateAsync: deleteLocation } = useLocationDelete();
+  const { success, info } = useNotification();
+  const [deleteErrors, setDeleteErrors] = React.useState<DeleteError[]>([]);
+  const { selectedRows } = useTableStore(state => ({
+    selectedRows: Object.keys(state.rowState)
+      .filter(id => state.rowState[id]?.isSelected)
+      .map(selectedId => data?.find(({ id }) => selectedId === id))
+      .filter(Boolean) as Location[],
+  }));
+
+  const deleteAction = () => {
+    const numberSelected = selectedRows.length;
+    if (selectedRows && numberSelected > 0) {
+      const errors: DeleteError[] = [];
+      Promise.all(
+        selectedRows.map(async location => {
+          await deleteLocation(location).then(data => {
```

Comment:
*cough* yes, not ideal.

Reply:
only for a bit till we can get a batch mutation!

With the batch mutation, I think it might also be easier to just throw an error within the mutation function and have `react-query` catch and set the errors in the `error` prop returned from `useMutation` also.

certainly something I've been neglecting so good to see some work put into error handling